tests/testsuite/freshness.rs

```diff
@@ -801,9 +792,9 @@ fn no_rebuild_when_rename_dir() {
     new.pop();
     new.push("bar");
     fs::rename(p.root(), &new).unwrap();
+    p.set_root(&new);
 
     p.cargo("build")
-        .cwd(&new)
```

Comment:
Using `.set_root` over `.cwd` is great. I think lots of tests would clean up nicely following this pattern.